sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriterV2.scala

```diff
@@ -343,3 +572,84 @@ trait CreateTableWriter[T] extends WriteConfigMethods[CreateTableWriter[T]] {
    */
   def tableProperty(property: String, value: String): CreateTableWriter[T]
 }
+
+case class WhenMatched[T] (dfWriter: DataFrameWriterV2[T], condition: Option[Expression]) {
```

Comment:
Some class docs for these classes should be good to add.

Reply:
Class docs are added. Thanks